src/Control/Functor/Linear/Internal/Class.hs

```diff
@@ -166,3 +208,133 @@ ap f x = f >>= (\f' -> fmap f' x)
 foldM :: forall m a b. Monad m => (b %1 -> a %1 -> m b) -> b %1 -> [a] %1 -> m b
 foldM _ i [] = return i
 foldM f i (x : xs) = f i x >>= \i' -> foldM f i' xs
+
+---------------
+-- Instances --
+---------------
+
+deriving via
+  Generically1 ((,) a)
+  instance
+    Functor ((,) a)
+
+deriving via
+  Generically1 ((,,) a b)
+  instance
+    Functor ((,,) a b)
+
+deriving via
+  Generically1 ((,,,) a b c)
+  instance
+    Functor ((,,,) a b c)
+
+deriving via
+  Generically1 ((,,,,) a b c d)
+  instance
+    Functor ((,,,,) a b c d)
+
+instance Monoid a => Monad ((,) a) where
+  (a, x) >>= f = go a (f x)
+    where
+      go :: a %1 -> (a, b) %1 -> (a, b)
+      go b1 (b2, y) = (b1 <> b2, y)
+
+deriving via
+  Generically1 (Sum.Sum f g)
+  instance
+    (Functor f, Functor g) => Functor (Sum.Sum f g)
+
+deriving via
+  Generically1 (Compose f g)
+  instance
+    (Functor f, Functor g) => Functor (Compose f g)
+
+------------------------
+-- Generics instances --
+------------------------
+
+instance (Generic1 f, Functor (Rep1 f)) => Functor (Generically1 f) where
+  fmap f = Generically1 . to1 . fmap f . from1 . unGenerically1
+
+-- True if the generic type does not contain 'Par1', i.e. it does not use its parameter.
+type family NoPar1 (f :: Type -> Type) :: Bool where
+  NoPar1 U1 = 'True
+  NoPar1 (K1 i v) = 'True
+  NoPar1 (l :*: r) = NoPar1 l && NoPar1 r
+  NoPar1 (l :+: r) = NoPar1 l && NoPar1 r
+  NoPar1 (l :.: r) = NoPar1 l || NoPar1 r
```

Comment:
Shouldn't that be
```suggestion
  NoPar1 (l :.: r) = NoPar1 l && NoPar1 r
```
?

Reply:
Nope. I don't actually remember if I wrote this or if @sjoerdvisscher did, but it's right. The composition only uses its parameter if *both* components do so.